src/main/java/edu/tamu/scholars/middleware/discovery/dto/DataNetworkDescriptor.java

```diff
@@ -0,0 +1,74 @@
+package edu.tamu.scholars.middleware.discovery.dto;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.solr.common.params.MapSolrParams;
+import org.apache.solr.common.params.SolrParams;
+
+public class DataNetworkDescriptor {
+
+    private final String id;
+
+    private final String dateField;
+
+    private final List<String> dataFields;
+
+    private final String typeFilter;
+
+    private DataNetworkDescriptor(String id, String dateField, List<String> dataFields, String typeFilter) {
+        super();
+        this.id = id;
+        this.dateField = dateField;
+        this.dataFields = dataFields;
+        this.typeFilter = typeFilter;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getDateField() {
+        return dateField;
+    }
+
+    public List<String> getDataFields() {
+        return dataFields;
+    }
+
+    public String getTypeFilter() {
+        return typeFilter;
+    }
+
+    public String getSort() {
+        return String.format("%s asc", dateField);
+    }
+
+    public String getFieldList() {
+        List<String> fields = new ArrayList<>(getDataFields());
+        fields.add(dateField);
+        return String.join(",", fields);
+    }
+
+    public String getFilterQuery() {
+        return String.format("syncIds:%s AND %s", id, typeFilter);
+    }
+
+    public SolrParams getSolrParams() {
+        final Map<String, String> queryParamMap = new HashMap<>();
+        queryParamMap.put("q", "*:*");
+        queryParamMap.put("rows", String.valueOf(Integer.MAX_VALUE));
+        queryParamMap.put("sort", getSort());
+        queryParamMap.put("fl", getFieldList());
+        queryParamMap.put("fq", getFilterQuery());
```

Comment:
I don't know what all is needed to be supported but my recent experience with Solr is that we should also support the `df` field for providing a default query field.

This would then allow for the query to be simply "*".
Of course, you can still do "*:*" to search all, bypassing whatever the `df` is set to.

I'm not sure what the requirements and expected functionality is so I do not consider this commit a review blocker at all.

I just felt this to be worth mentioning.

Reply:
For scholars I use the default search query as default and do not require it being specified. Didn't see any need at the time to switch between full-text search fields.